src/gui/torrentcontentmodel.cpp

```diff
@@ -91,14 +91,18 @@ namespace
         QIcon icon(const QFileInfo &info) const override
         {
             SHFILEINFO sfi = { 0 };
-            HRESULT hr = ::SHGetFileInfoW(info.absoluteFilePath().toStdWString().c_str(),
+            QString nativePath = Utils::Fs::toNativePath(info.absoluteFilePath());
```

Comment:
A user already reported that contents loading is slow on Windows, probably due to icons fetching. Thus we need icons cache on Windows too. So, creating a `CachingFileIconProvider` class seems to be a right choice, and then we should pass only extension to the actual icon fetching code.